### PR TITLE
RFC manipulate changes

### PR DESCRIPTION
**Pull request by JobJob**
So this is a proposal to:

1. remedy #12 
1. allow multiple values to be returned by manipulate

It doesn't display widgets by default, but adds a method to `display_dict and metadata for (::Tuple{Vararg{Union{Signal,Widget}}}` which will display the widgets and the signals. `Out[x]` fro the previous cell still holds the correct result somehow so that's cool.

Also if you return a N-tuple from manipulate it will create N signals.

### CODE REVIEW

File: src/IJulia/setup.jl [hunk 1]
```diff
@@ -305,5 +305,13 @@ function create_widget_signal{T<:Widget}(s::Signal{T})
     end
 end
 
+#Manipulate Helpers
+typealias ITypes Union{Signal,Widget}
+display_dict(x::Tuple{Vararg{ITypes}}) = begin
+    foreach(display,x)
+    Dict()
+end
+metadata(x::Tuple{Vararg{ITypes}}) = Dict()
```
Comment (shashi) [inline src/IJulia/setup.jl]: this is what metadata returns by default, why is this required?

File: src/manipulate.jl [hunk 2]
```diff
@@ -35,8 +43,11 @@ macro manipulate(expr)
...
-                    display_widgets(syms)...,
-                    esc(map_block(block, syms))),
-         map(make_widget, bindings)...)
+    Expr(:let,
+            Expr(:tuple,
+                map(sym->esc(sym), syms)...,
+                esc(Expr(:...,map_block(block, syms)))
+            ),
+            map(make_widget, bindings)...
+    )
```
Comment (shashi) [inline src/manipulate.jl]: Why not make this a composite type instead of a tuple? Overriding methods on `Vararg{IType}` looks hacky to me. Besides, that means we can't have anything other than IType things in the output.

Something like:

```julia
immutable Manipulate
    widgets::Dict # parameter name to widget mapping
    output::Signal
    layout::Function # Manipulate -> hbox / vbox (Can layout widgets using hbox and vbox) - has default, can be configued via @manipulate if necessary
end
```

could do the trick. Is it possible now to display _signals_ in an `hbox` or `vbox`? Then the default layout could just `vbox` the output below the widgets... Otherwise for now we could just return the output and let IJulia handle it as it now does.

File: src/IJulia/setup.jl (hunk 1 above)
Comment (JobJob) [inline src/IJulia/setup.jl]: Ahh cool, yep I don't think it's required

File: src/manipulate.jl (hunk 2 above)
Comment (JobJob) [inline src/manipulate.jl]: Yeah it is a bit hacky 😄 
> Is it possible now to display signals in an hbox or vbox?
Not right now, but haven't looked into what is required, haven't really looked into what Interact does on the js side either.

Comment (Rezoseansstudios) [inline src/manipulate.jl]: WHAT DE HELL?

Comment (Rezoseansstudios) [inline src/manipulate.jl]: WHAT DE HELL ARE YOU TALKING ABOUT RIGHT NOW!

**Comment by piever** [comment]
Closing as I don't think the issues this solves still exist in the new InteractBase implementation, feel free to reopen if I'm wrong and this is still relevant.
